### PR TITLE
Blog: The 8 part guide to better code review checklists

### PR DESCRIPTION
**Pull request by johnson-brad**
@featurepeek /blog/the-8-part-guide-to-better-code-review-checklists/

### CODE REVIEW

**Comment by johnson-brad** [comment]
Still need to pick out images for this one. Soliciting feedback on the revised draft in the meantime.

File: content/blog/the-8-part-guide-to-better-code-review-checklists/index.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# 2. Styling 🎨
+
+Set some basic styling rules for your team and ensure these are followed. Consistent code styling is essential to ensure future developers can understand code easily and work productively.
+
+Some basic rules:
+* Are variable names sensible and consistently capitalized?
+* Are there sufficiently descriptive comments throughout the code as required?
+* Are formatting preferences followed? For example, tabs or spaces, curly brace on the same line or new line, 80 character width or 120 character width?
+
+To save time, you could check these rules are followed with an automatic code linter.
```
Comment (jasonbarry) [inline content/blog/the-8-part-guide-to-better-code-review-checklists/index.md:53]: [Prettier](https://prettier.io) is a good example to link to for JavaScript projects.

File: content/blog/the-8-part-guide-to-better-code-review-checklists/index.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Again, if these standards are not met, stop checking the code and return it to the contributor for review.
+
+# 3. Security 🔐
+
+Set security standards for your project and check these are being followed religiously. The rules to be checked will vary depending on your project and organization, but some best practices are:
+
+* Run your project through vulnerability scanning solutions, like LGTM
```
Comment (jasonbarry) [inline content/blog/the-8-part-guide-to-better-code-review-checklists/index.md:61]: [LGTM](https://lgtm.com)

File: content/blog/the-8-part-guide-to-better-code-review-checklists/index.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+For example:
+* Can any of the code be substituted with a more performant library or a language native function?
+* Is there any debug or logging code that could be removed?
+* Is caching being used if applicable?
+* Are images and assets appropriately compressed?
+
+# 5. Testing 🧪
+
+Tests automatically check that code does what it should, making them a key part of the code review process.
```
Comment (jasonbarry) [inline content/blog/the-8-part-guide-to-better-code-review-checklists/index.md:81]: Would be good to mention [Jest](https://jestjs.io) for automated unit tests, and [Cypress](https://www.cypress.io) / [Reflect](https://reflect.run) for automated end-to-end tests.

File: content/blog/the-8-part-guide-to-better-code-review-checklists/index.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+* Don't hardcode credentials for testing or include secrets in the repository
+* Don't disclose too much information in error messages, which might give hints to an attacker
+* Ensure any database queries are parameterized
+* Don't trust user or client input blindly — your client or web application can be modified. For example, URL parameters that access resources should be checked.
+
+If you identify security issues in a code review process, stop and speak to the contributor. It might be a sign of a more serious issue in the project or a lack of training, both of which will need follow on interventions.
+
+# 4. Performance 🏎
+
+Is the project as performant as it could be, or are these obvious optimizations that might improve performance?
```
Comment (jasonbarry) [inline content/blog/the-8-part-guide-to-better-code-review-checklists/index.md:71]: Shout-out to [Chrome Lighthouse](https://developers.google.com/web/tools/lighthouse) (also called [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)) would be good in this section, as well as [DebugBear](https://www.debugbear.com) for frontend projects.

**Review by jasonbarry — COMMENTED** [review]
screenshot

There's a weird double-bullet in the ToC – I think the ToC assumes that h2's always follow h1's. You may need to change the "What is a code review" heading to be a h1.

<details><summary>Browser metadata</summary>


    Path:          /blog/the-8-part-guide-to-better-code-review-checklists/
    Browser:       Safari 13.1.2 on Mac OS 10.14.6
    Viewport:      1280 x 714 @2x
    Language:      en-us

    
</details>

[View on FeaturePeek](https://peek.run/mp4hz2g7) | [Mark as Resolved](https://dashboard.featurepeek.com/resolve/mp4hz2g7/lm4cz0/resolved)

**Pull request merged**